src/lib/merkle_ledger/test/test_mask.ml

```diff
@@ -633,6 +633,52 @@ module Make (Test : Test_intf) = struct
               ~message:"account in mask should be unchanged" ~expect:acct1
               (Mask.Attached.get attached_mask loc |> Option.value_exn) ) )
 
+  let () =
+    add_test "all_accounts_on_masks works on a chain" (fun () ->
+        Test.with_chain (fun base ~mask:m1_lazy ~mask2:m2_lazy ->
+            let num_accounts = 1 lsl min Test.depth 5 in
+            let num_layer_accounts = num_accounts / 3 in
+            let account_ids = Account_id.gen_accounts num_accounts in
+            let balances =
+              Quickcheck.random_value
+                (Quickcheck.Generator.list_with_length num_accounts Balance.gen)
+            in
+            let T = Account_id.eq in
+            let accounts =
+              List.map2_exn account_ids balances ~f:Account.create
```

Comment:
Nit: can we simplify all the accounts generation in this file with generators defined in 
`Mina_base.Account`? I'm fine as a follow up PR